src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/AbstractDocumentService.java

```diff
@@ -88,4 +108,63 @@ CaseDetails copyOf(CaseDetails caseDetails) {
             throw new IllegalStateException();
         }
     }
+
+    CaseDetails prepareNotificationLetter(CaseDetails caseDetails) {
+        Map<String, Object> caseData = caseDetails.getData();
+        Map addressToSendTo;
+
+        String ccdNumber = nullToEmpty((caseDetails.getId()));
+        String reference = "";
+        String addresseeName;
+        String applicantName = nullToEmpty((caseData.get(APPLICANT_FIRST_MIDDLE_NAME)))
+            + " " + nullToEmpty((caseData.get(APPLICANT_LAST_NAME)));
+        String respondentName = nullToEmpty((caseData.get(APP_RESPONDENT_FIRST_MIDDLE_NAME)))
+            + " " + nullToEmpty((caseData.get(APP_RESPONDENT_LAST_NAME)));
+
+        if (isApplicantRepresentedByASolicitor(caseData)) {
+            log.info("Applicant is represented by a solicitor");
+            reference = nullToEmpty((caseData.get(SOLICITOR_REFERENCE)));
+            addresseeName = nullToEmpty((caseData.get(SOLICITOR_NAME)));
+            addressToSendTo = (Map) caseData.get(APP_SOLICITOR_ADDRESS_CCD_FIELD);
+        } else {
+            log.info("Applicant is not represented by a solicitor");
+            addresseeName = applicantName;
+            addressToSendTo = (Map) caseData.get(APPLICANT_ADDRESS);
+        }
+
+        if (addressLineOneAndPostCodeAreBothNotEmpty(addressToSendTo)) {
+            Addressee addressee = Addressee.builder()
+                .name(addresseeName)
+                .formattedAddress(letterAddressHelper.formatAddressForLetterPrinting(addressToSendTo))
+                .build();
+
+            caseData.put("caseNumber", ccdNumber);
+            caseData.put("reference", reference);
+            caseData.put("addressee",  addressee);
+            caseData.put("letterDate", String.valueOf(LocalDate.now()));
+            caseData.put("applicantName", applicantName);
+            caseData.put("respondentName", respondentName);
+            caseData.put("ctscContactDetails", buildCtscContactDetails());
+
+        } else {
+            log.info("Failed to generate Notification Letter as not all required address details were present");
+            throw new IllegalArgumentException(
+                "Mandatory data missing from address when trying to generate Assigned To Judge Notification Letter");
+        }
+        return caseDetails;
+    }
+
+    CtscContactDetails buildCtscContactDetails() {
+
+        return CtscContactDetails.builder()
+            .serviceCentre("Courts and Tribunals Service Centre")
+            .careOf("c/o HMCTS Digital Financial Remedy")
```

Comment:
If not too much hassle, I think the "c/o " string should be in Docmosis (as optional).
That's because I saw other templates that only use the "HMCTS Digital Financial Remedy" part of this and that would mean either not reusing this object or having to remove the string in Docmosis.
Again, if it's too much hassle, few free not to do it.